R/spliceHelperFunctions.R

```diff
@@ -480,27 +484,39 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth=5, cutStartEnd=
 #' @param stand strand
 #' @noRd
 selectStartExonFromRangesList <- function(range, strand){
-  posStrandRng <- range[which(strand == "+")]
-  negStrandRng <- range[which(strand == "-")]
-  negStrandRng <- negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))]
-  startRng <- c(posStrandRng[which(start(posStrandRng) == min(start(posStrandRng)))],
-                negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))])
-  startRng <- startRng[sort(order(startRng)[names(range)])]
-  return (startRng)
+  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
+  largestExons <- as.numeric(cumsum(elementNROWS(range)))
+  startExonsSet <- c(1, largestExons[1:length(largestExons)-1]+1)
```

Comment:
maybe this would be simpler: `[-(length(lastExons))]`, currently you select the 0 element which is not clearly defined. You can also avoid this by using brackets `[1:(length(largestExons)-1)]`

Reply:
It is now changed to [-(length(lastExons))] in both select[Start/End]ExonFromRangesList() functions.